lib/jsonapi/view.ex

```diff
@@ -217,7 +217,7 @@ defmodule JSONAPI.View do
 
       def url_for(data, nil), do: "#{namespace()}/#{type()}/#{id(data)}"
 
-      def url_for(data, %Plug.Conn{} = conn) when is_list(data) do
+      def url_for(data, %Plug.Conn{} = conn) when is_list(data) or is_nil(data) do
```

Comment:
Why is this new guard necessary?

Reply:
or add possibility to return `%{data: null, etc...}` if render was called with `data = nil` . This must be the case specified in https://jsonapi.org/format/#document-top-level in the phrase 
```
Primary data MUST be either:

 - a single resource object, a single resource identifier object, or null, for requests that target single resources
- an array of resource objects, an array of resource identifier objects, or an empty array ([]), for requests that target resource collections
```